src/snowflake/snowpark/dataframe.py

```diff
@@ -1089,6 +1091,32 @@ def col(self, col_name: str) -> Column:
         else:
             return Column(self._resolve(col_name))
 
+    def col_regex(self, regex: str, case_sensitive: bool = False) -> Column:
+        """Selects column based on the column name specified as a regex and returns a ``Column`` reference to it.
+        Args:
+            regex: regular expression used to match columns
+        Examples:
+            >>> df = session.create_dataframe([[1, 2, 3, 4]]).to_df(["col1", "col2_a", "col2_b", "col3"])
+            >>> df.select(df.col_regex("col2_.*")).collect()
+            [Row(COL2_A=2, COL2_B=3)]
+
+        """
+        if not isinstance(regex, str):
+            raise ValueError(
+                f"regex provided to col_regex() must be string, got {type(regex)} with value of {regex} instead."
+            )
+        expressions = []
+        flag = 0 if case_sensitive else re.IGNORECASE
```

Comment:
What does 0 mean here?

Reply:
0 is the default value if you don't provide flag parameter, this act as a placeholder here